sharding/node/backend.go

```diff
@@ -92,49 +93,24 @@ func New(ctx *cli.Context) (*ShardEthereum, error) {
 }
 
 // Start the ShardEthereum service and kicks off the p2p and actor's main loop.
-func (s *ShardEthereum) Start() error {
+func (s *ShardEthereum) Start() {
 	log.Info("Starting sharding node")
-
-	services := make(map[reflect.Type]sharding.Service)
-	for _, constructor := range s.serviceFuncs {
-		// Create a new context for the particular service.
-		ctx := &sharding.ServiceContext{
-			Services: make(map[reflect.Type]sharding.Service),
-		}
-		// Copy needed for threaded access.
-		for kind, s := range services {
-			ctx.Services[kind] = s
-		}
-		// Construct and save the service.
-		service, err := constructor(ctx)
-		if err != nil {
-			return err
-		}
-		kind := reflect.TypeOf(service)
-		if _, exists := services[kind]; exists {
-			return fmt.Errorf("service already exists: %v", kind)
-		}
-		services[kind] = service
-	}
-
-	// Start each of the services.
-	started := []reflect.Type{}
-	for kind, service := range services {
-		// Start the next service, stopping all previous upon failure.
+	for kind, service := range s.services {
+		// Start the next service.
 		if err := service.Start(); err != nil {
-			for _, kind := range started {
-				services[kind].Stop()
-			}
-			return err
+			log.Error(fmt.Sprintf("Could not start service: %v, %v", kind, err))
 		}
-		// Mark the service started for potential cleanup.
-		started = append(started, kind)
 	}
-	return nil
 }
 
 // Close handles graceful shutdown of the system.
 func (s *ShardEthereum) Close() error {
+	for kind, service := range s.services {
+		if err := service.Stop(); err != nil {
+			return fmt.Errorf(fmt.Sprintf("Could not stop service: %v", kind))
```

Comment:
So if the first service fails to stop, the rest are not stopped? 
What if you have persistent database connections or disk write operations that you would really like to stop gracefully?